File: Data Structures & Algorithms/same-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        self.ret = True
        def dfs(nodeP, nodeQ):
            if (nodeP and not nodeQ) or (nodeQ and not nodeP):
                self.ret = False
                return
            if (not nodeP and not nodeQ):
                return

            if (nodeP.val != nodeQ.val):
                self.ret = False
                return
            
            dfs(nodeP.left, nodeQ.left)
            dfs(nodeP.right, nodeQ.right)

            return

        dfs(p, q)
        return self.ret

File: Data Structures & Algorithms/reverse-bits/submission-1.cpp
using namespace std;

class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t ret {0};
        uint32_t curr = 1 << 31;
        cout << curr << endl;
        for (int i = 0; i < 32; i++) {
            if (n % 2 == 1) {
                ret += curr;
                cout << ret << endl;
            }
            curr = curr >> 1;
            n = n >> 1;
        }
        return ret;
    }
};
